rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -0,0 +1,350 @@
+    severity:'NOTICE',\
+    chain"
+        SecRule MATCHED_VARS "@rx (?:runtime|processbuilder|clonetransformer|forclosure|instantiatefactory|instantiatetransformer|invokertransformer|prototypeclonefactory|prototypeserializationfactory|whileclosure)" \
+            "t:base64Decode,t:lowercase,\
+            setvar:'tx.msg=%{rule.msg}',\
+            setvar:tx.rce_score=+%{tx.notice_anomaly_score},\
+            setvar:tx.anomaly_score=+%{tx.notice_anomaly_score},\
+            setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{matched_var}'"
+
+# Looking for possibly RCE vulnerable java classes,
+# anomaly score set to notice as both conditions may indicate a possibly vulnerable application
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML:/*|XML://@* \
+    "@rx (?:clonetransformer|forclosure|instantiatefactory|instantiatetransformer|invokertransformer|prototypeclonefactory|prototypeserializationfactory|whileclosure)" \
+    "id:944330,\
+    phase:2,\
+    block,\
+    t:lowercase,\
+    log,\
+    msg:'Probable vulnerable java class in use',\
+    logdata:'Matched Data: %{MATCHED_VAR} found within %{MATCHED_VAR_NAME}',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    severity:'NOTICE',\
+    setvar:'tx.msg=%{rule.msg}',\
+    setvar:tx.rce_score=+%{tx.notice_anomaly_score},\
+    setvar:tx.anomaly_score=+%{tx.notice_anomaly_score},\
+    setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{matched_var}'"
+
+# Magic bytes detected and payload included possibly RCE vulnerable classess detected and process execution methods detected
+# anomaly score set to critical as all conditions indicate the request try to perform RCE.
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML:/*|XML://@* \
+    "@rx (?:clonetransformer|forclosure|instantiatefactory|instantiatetransformer|invokertransformer|prototypeclonefactory|prototypeserializationfactory|whileclosure)" \
+    "id:944340,\
+    phase:2,\
+    block,\
+    t:lowercase,\
+    log,\
+    msg:'Remote Command Execution: Java serialization (CVE-2015-5842)',\
+    logdata:'Matched Data: %{MATCHED_VAR} found within %{MATCHED_VAR_NAME}',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    severity:'CRITICAL',\
+    chain"
+        SecRule MATCHED_VARS "@rx (?:runtime|processbuilder)" \
```

Comment:
This rule seems pretty specific and not prone to FP, what about moving it to a lower Paranoia Level like 2?